### PR TITLE
Add optional separate throttling time for apiV4 to ghproxy algorithm

### PR DESCRIPTION
**Pull request by jmguzik**
This commit adds an optional, separate setting of throttling time for apiV4 to the ghproxy algorithm
implemented as the part of #24847. Deployed algorithm causes tide slowdown if non-GET time throttling is
set to a high value. However, lowering the value to a level of 300ms seems to be enough for tide not
to get rate limit throttled and keep up with sync work. Since most of the apiV4 uses POST as an HTTP method,
it is possible that Github api allows different timings for non-GET methods in v3 and v4. Tide not being
throttled (and using v4 for these intensive syncs) on the 300 ms setting is good evidence.

Signed-off-by: Jakub Guzik <jguzik@redhat.com>

### CODE REVIEW

**Comment by jmguzik** [comment]
To sum up the above description: an optional algorithm recently added helped tide not to catch secondary rate limits in our deployment. However, a setting of 300ms seems not enough for `POST` and others in API v3. So, I propose to distinguish v3 and v4 non-GET throttling timings.

Additionally, I did some cleaning not to pass 3 different uints to a bunch of functions. I created a struct to do that.

File: ghproxy/ghcache/ghcache.go [hunk 1]
```diff
@@ -85,6 +85,39 @@ const (
...
+	throttlingTimeV4 uint
+	// ThrottlingTimeForGET is applied for all GET request methods for apiV3 and apiV4
+	throttlingTimeForGET uint
+}
+
+func (rtt *RequestThrottlingTimes) isEnabled() bool {
+	if rtt.throttlingTime > 0 && rtt.throttlingTimeForGET > 0 {
+		return true
+	}
+	return false
```
Comment (petr-muller) [inline ghproxy/ghcache/ghcache.go:102]: Maybe just `return rtt.throttlingTime > 0 && rtt.throttlingTimeForGET > 0`?

**Review by petr-muller — APPROVED** [review]
LGTM up to one nit

/cc @alvaroaleman @cjwagner 
As a followup to the discussion in the previous throttling PR, after trying the throttling feature in production, we can report very positive results actually:

1) We have virtually eliminated all rate limiting in Tide (which mostly does v4 `POST`s)
2) Overall the situation improved, we see fewer rate limiting events and the periods are shorter - the remaining ones seem to be connected to crier `POST`s
3) We haven't observed any periods where we could not "keep up" with the GH API demand. The `POSTs` done by `crier` are observably slower (you can observe statuses appearing in groups of four or five when you do e.g. `/test all` or open a PR.) but not to the point affecting UX: you simply have all contexts submitted in several seconds instead of all at once.

File: ghproxy/ghcache/ghcache.go (hunk 1 above)
Comment (jmguzik) [reply ghproxy/ghcache/ghcache.go]: Yes, that was stupid. Corrected.

**Review by cjwagner — APPROVED** [review]

**Pull request merged**